maplestory-cursor: init at 1.0.0

Liam-Weitzel opened the pull request:
My first nix package, a maplestory inspired cursor theme. Works on both x11 and wayland. A fork of the popular banana-cursor.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Liam-Weitzel committed:
maintainers: add liam-w

Liam-Weitzel committed:
maplestory-cursor: init at 1.0.0

langsjo commented on pkgs/by-name/ma/maplestory-cursor/package.nix:
```diff
@@ -0,0 +1,51 @@
+  src = fetchFromGitHub {
+    owner = "Liam-Weitzel";
+    repo = "maplestory-cursor";
+    rev = "736862bb07c4984e1b287ef26f9838177353abee";
```
```suggestion
    tag = "v${finalAttrs.version}";
```

langsjo commented on pkgs/by-name/ma/maplestory-cursor/package.nix:
```diff
@@ -0,0 +1,51 @@
+  '';
+
+  meta = {
+    description = "A Maplestory cursor theme for x11 & wayland";
```
```suggestion
    description = "Maplestory cursor theme for x11 & wayland";
```

Liam-Weitzel replied:
done

langsjo commented:
And welcome to nixpkgs! These are just some small changes to follow conventions, but otherwise looks good in my eyes

Iamanaws commented:
Good job addressing issues.

As I mentioned in https://github.com/NixOS/nixpkgs/pull/423912,

I still want to note that this package may not have sufficient relevance.

[Before adding a new package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package)

> How realistic is it that it will be used by other people? It's good that nixpkgs caters to various niches, but if it's a niche of 5 people it's probably too small. A good estimate is checking upstream issues and pull requests, or other software repositories. Library packages should have at least one dependent.


Checkout [NUR](https://github.com/nix-community/NUR) for personal packages

Liam-Weitzel commented on pkgs/by-name/ma/maplestory-cursor/package.nix:
```diff
@@ -0,0 +1,51 @@
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "maplestory-cursor";
+  version = "v1.0.0";
```
done

Liam-Weitzel commented:
> Good job addressing issues.
> 
> As I mentioned in #423912,
> 
> I still want to note that this package may not have sufficient relevance.
> 
> [Before adding a new package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package)
> 
> > How realistic is it that it will be used by other people? It's good that nixpkgs caters to various niches, but if it's a niche of 5 people it's probably too small. A good estimate is checking upstream issues and pull requests, or other software repositories. Library packages should have at least one dependent.
> 
> Checkout [NUR](https://github.com/nix-community/NUR) for personal packages

Hi, yes, apologies for making a new PR. I usually work collaboratively on gitlab. That is a valid concern and honestly I don't know if more than 5 people will use this. The parent package 'banana-cursor' has had some success (281 stars), and this plugin builds on top of it, so it's fair to say the audience is quite small. I submitted it here because it seemed like a natural extension of what's already in nixpkgs & i'd hoped others would appreciate it. But I totally understand if it doesn't meet the bar for inclusion.

Liam-Weitzel commented:
> Hi, yes, apologies for making a new PR. I usually work collaboratively on gitlab. That is a valid concern and honestly I don't know if more than 5 people will use this. The parent package 'banana-cursor' has had some success (281 stars), and this plugin builds on top of it, so it's fair to say the audience is quite small. I submitted it here because it seemed like a natural extension of what's already in nixpkgs & i'd hoped others would appreciate it. But I totally understand if it doesn't meet the bar for inclusion.

That said, I’ve noticed that there are other packages recently merged into nixpkgs that seem similarly niche or limited in audience - for example, #422351, which appears to be a classroom project with a fairly minimal upstream presence (12 stars). I think it would be super cool if this package was merged and I'm sure I can convince my 12 friends to use this cursor theme 😜

Liam-Weitzel commented:
> All checks have run fine. The maintainers on Darwin checks have been skip as this is Linux only, nothing to worry about.
Nice! What does the usual process look like for the final merge?

Iamanaws commented:
> Nice! What does the usual process look like for the final merge?

Usually, like a long wait until a committer give it a look. If running with luck you can get it merge in like 2 weeks, but it's been more than that, so this one seems like the long wait ones. (from some weeks to a few months 🥲)

Too many PRs and too little committers. But hopefully you will get it soon
